runtime/resetting_histogram.go

```diff
@@ -0,0 +1,305 @@
+
+// Clear is a no-op.
+func (NilResettingHistogram) Clear() {}
+
+func (NilResettingHistogram) Count() int64 {
+	panic("Count called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) Variance() float64 {
+	panic("Count called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) Min() int64 {
+	panic("Min called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) Max() int64 {
+	panic("Max called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) Sum() int64 {
+	panic("Sum called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) StdDev() float64 {
+	panic("StdDev called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) Sample() Sample {
+	panic("Sample called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) Percentiles([]float64) []float64 {
+	panic("Percentiles called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) Percentile(float64) float64 {
+	panic("Percentiles called on a NilResettingHistogram")
+}
+
+func (NilResettingHistogram) Mean() float64 {
+	panic("Mean called on a NilResettingHistogram")
+}
+
+// StandardResettingHistogram is used for storing aggregated values for timers, which are reset on every flush interval.
+type StandardResettingHistogram struct {
+	values []int64
+	mutex  sync.Mutex
+}
+
+func (t *StandardResettingHistogram) Count() int64 {
+	panic("Count called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) Max() int64 {
+	panic("Max called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) Min() int64 {
+	panic("Min called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) StdDev() float64 {
+	panic("StdDev called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) Variance() float64 {
+	panic("Variance called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) Sum() int64 {
+	panic("Sum called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) Sample() Sample {
+	panic("Sample called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) Percentiles([]float64) []float64 {
+	panic("Percentiles called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) Percentile(float64) float64 {
+	panic("Percentile called on a StandardResettingHistogram")
+}
+
+func (t *StandardResettingHistogram) Mean() float64 {
+	panic("Mean called on a StandardResettingHistogram")
+}
+
+// Values returns a slice with all measurements.
+func (t *StandardResettingHistogram) Values() []int64 {
+	return t.values
+}
+
+// Snapshot resets the timer and returns a read-only copy of its sorted contents.
+func (t *StandardResettingHistogram) Snapshot() Histogram {
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+	currentValues := t.values
+	t.values = make([]int64, 0, InitialResettingHistogramSliceCap)
+
+	sort.Sort(Int64Slice(currentValues))
```

Comment:
We could avoid the single-purpose type below.

```suggestion
	sort.Slice(currentValues, func(i, j int) bool { currentValues[i] < currentValues[j] })
```

Reply:
I agree, this was legacy pre Go1.8 code, your suggestion is better now!